core_java/grep/README.md

```diff
@@ -0,0 +1,49 @@
+# Java Grep Application
+
+## Introduction
+Grep command in Linux is used to search text or searches the given files for lines containing a match to the given strings or words. It outputs the matching lines, by default.
+Java Grep Application is a basic implementation of the same in Java. When it is run, it recursively searches all files in the specified directory and traverse line by line to find matches with a given regular expression. The output matched lines are then saved and written in a new file created by user.
+The purpose of this project is to thoroughly learn Core Java and Java 8 features, especially Lambda Expressions and Stream APIs. It also introduced us to IntelliJ IDEA IDE and Maven. Maven built and managed this app which is a build automation tool and Project Object Model (POM).
+
+
+## Usage
+To use this application, we need to specify three arguments described below:
+* _regex_ - Regular expression which will be searched for in all files in the specified directory.
+* _rootPath_ - Tt is a path to the directory that will be searched recursively. All the sub directories of this directory will also be searched.
+* _outFile_ - Output file where all the matched lines will be saved.
+For example - "SampleRegex" is the text that needs to be searched. The app would search the *rootPath*, in this case, `home/centos/dev`  directory for any lines that contain text SampleRegex (__regex - .*SampleText.*__). Once the matches are found, it would write the list of matched lines to *OutFile*, in this case, `home\centos\grep.out` 
+`.*SampleRegex.* home/centos/dev home/centos/grep.out`
+
+## Pseudocode
+High level description of working of JavaGrep functions:
+
+```
+matchedLines = []
+for file in listFilesRecursively(rootDir)
+        for line in readLines(file)
+        if containsPattern(line)
+                matchedLines.add(line)
+        writeToFile(matchedLines)
+```
+
+
+Both of our implementations - JavaGrepImp and JavaGrepLambda uses JavaGrep function.
+We have implemented the above workflow in process method().
+Description of methods in the workfow:
+* _listFilesRecursively()_ - traverse the specified root directory recursively and generate list of files in the directory.
```

Comment:
When you describe the signature of an API, it's suggested to include function name, parameters and return value.

Reply:
README updated.